### DISCUSSION TITLE
Request-FalconToken Invalid URI

### DISCUSSION THREAD

**Discussion by Yozhika** [CrowdStrike/psfalcon#259] [Q&A] [answered]
Hi,
I'm trying to request token and why do I get this? Invalid URI - is it the URL for Crowdstrike? So Powershell must be using a URL to go to Crowdstrike and it's invalid. Where do I check what URL Powershell is trying to use? 

PS C:\Users\redacted> Request-FalconToken -ClientId 'xxx' -ClientSecret 'xxx' -Cloud us-2
Exception calling ".ctor" with "2" argument(s): "Invalid URI: The hostname could not be parsed."
At C:\Users\redacted\redacted\Documents\WindowsPowerShell\Modules\PSFalcon\2.2.3\Class\Class.ps1:38 char:17
+ ...             $Message = [System.Net.Http.HttpRequestMessage]::New($Par ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : UriFormatException

When I run Powershell as admin, I get a similar message, but it's more straightforward: 

Write-Result : [{"code":403,"message":"Failed to issue access token - Not Authorized"}]
At C:\Users\redacted\redacted\Documents\WindowsPowerShell\Modules\psfalcon\2.2.3\Public\oauth2.ps1:168 char:27
+                 $Result = Write-Result $Request
+                           ~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidResult: (System.Threadin...esponseMessage]:Task`1) [Write-Result], Exception
    + FullyQualifiedErrorId : c15ebdb4-2f85-4243-b196-111be2f764a7,Write-Result

**Comment by bk-cs** [comment, marked as answer]
The error you received looks like it's due to the module not properly importing. It looks like your PowerShell module path is in OneDrive which could cause import problems.

Did you `Import-Module -Name PSFalcon` before attempting the token request?

If the import doesn't work, try a [manual install](https://github.com/crowdStrike/psfalcon/wiki/Installation,-Upgrade-and-Removal#download-from-github).

**Reply by Yozhika** [reply]
You were right. I did import it , but something must have been off, because I just tried again and I get no error.
Thanks
